packages/lodestar/src/chain/regen/regen.ts

```diff
@@ -29,6 +29,7 @@ export class StateRegenerator implements IStateRegenerator {
   private checkpointStateCache: CheckpointStateCache;
   private db: IBeaconDb;
   private metrics: IMetrics | null;
+  private rmetricsMap: Map<string, IRegenFnMetrics>;
```

Comment:
This Map should not be necessary. You can find alternative patterns, like passing the metrics to the state cache constructor